dev/com.ibm.ws.session.cache/src/com/ibm/ws/session/store/cache/CacheHashMap.java

```diff
@@ -896,22 +1124,104 @@ protected BackedSession readFromExternal(String id) {
-            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled())
-                Tr.debug(this, tc, "remove session properties", propIds);
             for (String propId : propIds) {
                 String propertyKey = createSessionPropertyKey(id, propId);
-                sessionPropertyCache.remove(propertyKey);
+
+                if (trace && tc.isDebugEnabled())
+                    tcInvoke(tcSessionAttrCache, "remove", propertyKey);
+
+                sessionAttributeCache.remove(propertyKey);
+
+                if (trace && tc.isDebugEnabled())
+                    tcReturn(tcSessionAttrCache, "remove");
+            }
+        }
+    }
+
+    /**
+     * Generates a trace string of the form,
+     * ==> Instance12345678.methName arg0, arg1, byte[22]: DeserializedFormOfArg2Bytes
+     * 
+     * @param instance instance
+     * @param methName method name
+     * @param args method arguments
+     */
+    @Trivial
+    static final void tcInvoke(String instance, String methName, Object... args) {
+        StringBuilder b = new StringBuilder("==> ").append(instance);
+        if (methName.charAt(0) != '_')
+            b.append('.');
+        b.append(methName).append(' ');
+        for (int i = 0; i < args.length; i++) {
+            if (i > 0)
+                b.append(", ");
+            if (args[i] instanceof byte[]) {
+                byte[] bytes = (byte[]) args[i];
+                b.append("byte[").append(bytes.length).append("]");
+                if (++i < args.length && bytes.length < 1000)
+                    b.append(": ").append(args[i]);
+            } else {
+                if (args[i] instanceof Collection)
+                    b.append(EOLN).append(' ');
+                b.append(args[i]);
+            }
+        }
+        Tr.debug(tc, b.toString());
+    }
+
+    /**
+     * Generates a trace string of the form,
+     * <== Instance12345678.methName byte[100]: DeserializedFormOfResult
+     *
+     * @param instance instance
+     * @param methName method name
+     * @param result first argument is the result or bytes representing the result. If the first argument is bytes, then the second argument is the result.
+     */
+    @Trivial
+    static final void tcReturn(String instance, String methName, Object... result) {
+        StringBuilder b = new StringBuilder("<== ").append(instance);
+        if (methName.charAt(0) != '_')
+            b.append('.');
+        b.append(methName).append(' ');
+        if (result == null) {
+            b.append("null");
+        } else if (result.length > 0) {
+            byte[] bytes;
+            if (result[0] instanceof byte[]) {
+                bytes = (byte[]) result[0];
+                b.append("byte[").append(bytes.length).append("]");
+            } else {
+                bytes = null;
+                if (result[0] instanceof Collection)
+                    b.append(EOLN).append(' ');
+                b.append(result == null ? null : result[0]);
```

Comment:
It isn't hurting anything, but I don't think this null check is needed, since it is in the else block of an if statement checking whether the result was null.

Reply:
removed the redundant null check.
I added a commit with this and the other changes above